packages/gatsby/cache-dir/loader.js

```diff
@@ -232,28 +232,31 @@ export class BaseLoader {
 
       const finalResult = {}
 
-      const componentChunkPromise = this.loadComponent(componentChunkName).then(
-        component => {
-          finalResult.createdAt = new Date()
-          let pageResources
-          if (!component) {
-            finalResult.status = PageResourceStatus.Error
-          } else {
-            finalResult.status = PageResourceStatus.Success
-            if (result.notFound === true) {
-              finalResult.notFound = true
-            }
-            pageData = Object.assign(pageData, {
-              webpackCompilationHash: allData[0]
-                ? allData[0].webpackCompilationHash
-                : ``,
-            })
-            pageResources = toPageResources(pageData, component)
+      console.log({ pageData })
+      const componentChunkPromise = this.loadComponent(
+        componentChunkName,
+        pageData.path
+      ).then(component => {
```

Comment:
What happens when `loadComponent` throws an error? Or when the `then` callback does? I think this goes to most promises here. Is there a general uncaught rejection handler in place?

Reply:
I don't think so. I know we've talked about adding a general error boundary for this stuff in React land but that work hasn't been done yet AFAIK. @wardpeet or @pieh might know.

Reply:
ProdLoader's `loadComponent` method current implementation catches rejection internally and it resolve to `null` - https://github.com/gatsbyjs/gatsby/blob/ae2c3dab9fe6b26a968b7762892243914ee10a6b/packages/gatsby/cache-dir/loader.js#L439-L447

There should be some `null` checks by whatever call `loadComponent` method, and I think behaviour should be similar as when user tries to navigate to a page that can't load `page-data.json` file - https://github.com/gatsbyjs/gatsby/blob/ae2c3dab9fe6b26a968b7762892243914ee10a6b/packages/gatsby/cache-dir/loader.js#L476-L493

(above code is done for the most part as resource loading resilience attempt, to try to load the page in traditional way if resources are blocked or missing)

DevLoader (speaking about `master` branch, not this PR as I didn't follow changes yet) doesn't have anything like that  but it also doesn't do any async work, because we don't split chunks in develop and everything load at once, so if "module loading" would cause errors it will likely break entire runtime at the time we import `sync-requires` (so https://github.com/gatsbyjs/gatsby/blob/ae2c3dab9fe6b26a968b7762892243914ee10a6b/packages/gatsby/cache-dir/app.js#L11 )?